scripts/gulp/plugins/util/docgen.ts

```diff
@@ -87,7 +87,15 @@ const defaultOptions: ts.CompilerOptions = {
   allowUnreachableCode: true,
 };
 
-const reactComponentSymbolNames = ['StatelessComponent', 'Stateless', 'StyledComponentClass', 'FunctionComponent'];
+const reactComponentSymbolNames = [
+  'StatelessComponent',
+  'Stateless',
+  'StyledComponentClass',
+  'FunctionComponent',
+
+  // magic for ComponentWithAs
+  '__type',
```

Comment:
What was this meant to do? This addition of custom logic makes it harder to move back to the official `react-docgen-typescript`. :( @layershifter or maybe @mnajdova ?

Reply:
`react-typescript-docgen` uses TS AST, for some reason it returns `__type` from `.getSymbolName()`. I tested it with the latest `react-docgen-typescript` and in this case it works 🤔 But there is another blocking issue: https://github.com/styleguidist/react-docgen-typescript/issues/246

Reply:
Got it, in that case we can keep the fork for now.